src/util/async/Operation.hpp

```diff
@@ -181,4 +186,58 @@ using Operation = impl::BasicOperation<Outcome<RetType>>;
 template <typename CtxType, typename OpType>
 using ScheduledOperation = impl::BasicScheduledOperation<CtxType, OpType>;
 
+/**
+ * @brief The `future` side of async operations that automatically repeat until aborted
+ *
+ * @note The current implementation requires the user provided function to return void and to take no arguments. There
+ * is also no mechanism to request the repeating task to stop from inside of the user provided block of code.
+ *
+ * @tparam CtxType The type of the execution context
+ */
+template <typename CtxType>
+class RepeatingOperation {
+    using DataType = std::expected<void, ExecutionError>;
+    using StopSourceType = typename CtxType::StopSource;
+
+    std::unique_ptr<util::Repeat> repeat_;
```

Comment:
Why unique_ptr?

Reply:
Repeat is not movable. But let me try with your PR #1775